### DISCUSSION TITLE
Which ports does a self hosted runner need?

### DISCUSSION THREAD

**Discussion by dirkhh** [community/community#26630] [Actions]
<p>I’m playing with the idea of self-hosting a runner since there is one Action that I need to take (fairly rarely) that runs more than 6h on a 2 core system. Not ideal, but oh well.</p>
<p>In order to minimize the risk I want to completely isolate that runner in its own DMZ, but it’s unclear how the communication between the runner and GitHub works. Is there a document that describes this beyond “it uses https to communiate” - do I need any inbound ports? Or is all communication runner-initiated and all I need is outbound tcp/443?</p>
<p>Any additional documentation on this would be appreciated.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Thanks for your feedback!</p>
<p>Self-hosted runner is a new feature of github action, typical usage is documented in this <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/hosting-your-own-runners" rel="nofollow noopener">link.</a></p>
<p>The runners connect back to GitHub, no need for inbound firewall holes. (Outbound https 443 is all that’s needed.)</p>

**Comment by rivadaviam** [comment]
<p>So only outbound is needed from the runner?<br>
The runner is pooling for new job to run?<br>
From the security point of view is awesome, can you confirm this please?</p>
<p>Thank you.</p>

**Comment by jkruse14** [comment]
Self-hosted runners use HTTPS long poll with a connection timeout of 50 seconds. When that timeout occurs, a new connection is opened. This means you do not need to allow GitHub to make inbound connections to your runner. More [here](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners#communication-between-self-hosted-runners-and-github)
